File: Source/Nexus/src/Nexus/Core/Input.cpp
#include "nxpch.h"
#include "Input.h"
#include "GLFW/glfw3.h"

namespace Nexus::Input
{
	GLFWwindow* m_window = nullptr;
}

void Nexus::Input::SetContextWindow(const Window& window)
{
	m_window = window.handle;
}

bool Nexus::Input::IsKeyPressed(uint16_t key)
{
	auto state = glfwGetKey(m_window, key);
	return state == GLFW_PRESS || state == GLFW_REPEAT;
}

bool Nexus::Input::IsMouseButtonPressed(uint16_t button)
{
	return glfwGetMouseButton(m_window, button) == GLFW_PRESS;
}

std::pair<float, float> Nexus::Input::GetMouseCursorPosition()
{
	double x, y;
	glfwGetCursorPos(m_window, &x, &y);
	return { (float)x,(float)y };
}

File: Source/NexusScriptCore/Source/Nexus/Core/Input.cs
using System;

namespace Nexus
{
    public class Input
    {
        static public bool IsKeyPressed(KeyCode keyCode)
        {
            return InternalCalls.Input_IsKeyPressed((UInt16)keyCode);
        }
        static public bool IsMouseButtonPressed(MouseCode mouseCode)
        {
            return InternalCalls.Input_IsMouseButtonPressed((UInt16)mouseCode);
        }
    }
}
